eth/tracers/live/supply.go

```diff
@@ -0,0 +1,266 @@
+package live
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"runtime"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/eth/tracers/directory"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+func init() {
+	directory.LiveDirectory.Register("supply", newSupply)
+}
+
+type SupplyInfo struct {
+	Delta       *big.Int `json:"delta"`
+	Reward      *big.Int `json:"reward"`
+	Withdrawals *big.Int `json:"withdrawals"`
+	Burn        *big.Int `json:"burn"`
+
+	// Block info
+	Number     uint64      `json:"blockNumber"`
+	Hash       common.Hash `json:"hash"`
+	ParentHash common.Hash `json:"parentHash"`
+}
+
+type supplyTxCallstack struct {
+	calls []supplyTxCallstack
+	burn  *big.Int
+}
+
+type Supply struct {
+	delta       SupplyInfo
+	txCallstack []supplyTxCallstack // Callstack for current transaction
```

Comment:
@s1na for handling the **selfdestruct** to itself case, I used a similar logic to what we have at callTracer for internal transactions. This way, later we can handle more stuff on inner transactions if needed. What do you think?

Reply:
Yes that's how I would have done it as well.